searchlib/src/vespa/searchlib/attribute/attribute_blueprint_factory.cpp

```diff
@@ -633,7 +633,11 @@ class DirectAttributeBlueprint : public queryeval::SimpleLeafBlueprint
                 return bitvector_iterator;
             }
         }
-        return std::make_unique<queryeval::DocumentWeightSearchIterator>(*tfmda[0], _attr, _dict_entry);
+        if (_attr.has_weight_iterator(_dict_entry.posting_idx)) {
+            return std::make_unique<queryeval::DocumentWeightSearchIterator>(*tfmda[0], _attr, _dict_entry);
+        } else {
+            return _attr.make_bitvector_iterator(_dict_entry.posting_idx, get_docid_limit(), *tfmda[0], strict);
```

Comment:
`tfmda[0]->populate_fixed()` is not called when using bitvector iterator
but it is called when using `DocumentWeightSearchIterator`. For a filter field that is used by ranking, this might matter, depending on what is used by the feature executor.

Reply:
Good point.